Source/ablastr/fields/IntegratedGreenFunctionSolver.cpp

```diff
@@ -145,57 +147,71 @@ computePhiIGF ( amrex::MultiFab const & rho,
-                amrex::Real const z = k0*dz;
-
-#if !defined(ABLASTR_USE_HEFFTE)
-                // Without distributed FFTs (i.e. without heFFTe):
-                amrex::Real const G_value = SumOfIntegratedPotential(x     , y     , z     , dx, dy, dz);
-                tmp_G_arr(i,j,k) = G_value;
-                // Fill the rest of the array by periodicity
-                if (i0>0) {tmp_G_arr(hi[0]+1-i0, j         , k         ) = G_value;}
-                if (j0>0) {tmp_G_arr(i         , hi[1]+1-j0, k         ) = G_value;}
-                if (k0>0) {tmp_G_arr(i         , j         , hi[2]+1-k0) = G_value;}
-                if ((i0>0)&&(j0>0)) {tmp_G_arr(hi[0]+1-i0, hi[1]+1-j0, k         ) = G_value;}
-                if ((j0>0)&&(k0>0)) {tmp_G_arr(i         , hi[1]+1-j0, hi[2]+1-k0) = G_value;}
-                if ((i0>0)&&(k0>0)) {tmp_G_arr(hi[0]+1-i0, j         , hi[2]+1-k0) = G_value;}
-                if ((i0>0)&&(j0>0)&&(k0>0)) {tmp_G_arr(hi[0]+1-i0, hi[1]+1-j0, hi[2]+1-k0) = G_value;}
-#else
-                // With distributed FFTs (i.e. with heFFTe):
-                amrex::Real x_hi = dx*(hi[0]+2);
-                amrex::Real y_hi = dy*(hi[1]+2);
-                amrex::Real z_hi = dz*(hi[2]+2);
-                if ((i0< nx)&&(j0< ny)&&(k0< nz)) { tmp_G_arr(i,j,k) = SumOfIntegratedPotential(x     , y     , z     , dx, dy, dz); }
-                if ((i0< nx)&&(j0> ny)&&(k0< nz)) { tmp_G_arr(i,j,k) = SumOfIntegratedPotential(x     , y_hi-y, z     , dx, dy, dz); }
-                if ((i0< nx)&&(j0< ny)&&(k0> nz)) { tmp_G_arr(i,j,k) = SumOfIntegratedPotential(x     , y     , z_hi-z, dx, dy, dz); }
-                if ((i0> nx)&&(j0> ny)&&(k0< nz)) { tmp_G_arr(i,j,k) = SumOfIntegratedPotential(x_hi-x, y_hi-y, z     , dx, dy, dz); }
-                if ((i0< nx)&&(j0> ny)&&(k0> nz)) { tmp_G_arr(i,j,k) = SumOfIntegratedPotential(x     , y_hi-y, z_hi-z, dx, dy, dz); }
-                if ((i0> nx)&&(j0< ny)&&(k0> nz)) { tmp_G_arr(i,j,k) = SumOfIntegratedPotential(x_hi-x, y     , z_hi-z, dx, dy, dz); }
-                if ((i0> nx)&&(j0> ny)&&(k0> nz)) { tmp_G_arr(i,j,k) = SumOfIntegratedPotential(x_hi-x, y_hi-y, z_hi-z, dx, dy, dz); }
-                if ((i0> nx)&&(j0< ny)&&(k0< nz)) { tmp_G_arr(i,j,k) = SumOfIntegratedPotential(x_hi-x, y     , z     , dx, dy, dz); }
+
+                if (is_2d_slices) {
+                    // 2D
+                    if ((i0< nx)&&(j0< ny)) { tmp_G_arr(i,j,k) = SumOfIntegratedPotential2D(x,      y,      dx, dy); }
+                    if ((i0< nx)&&(j0> ny)) { tmp_G_arr(i,j,k) = SumOfIntegratedPotential2D(x,      y_hi-y, dx, dy); }
+                    if ((i0> nx)&&(j0> ny)) { tmp_G_arr(i,j,k) = SumOfIntegratedPotential2D(x_hi-x, y_hi-y, dx, dy); }
+                    if ((i0> nx)&&(j0< ny)) { tmp_G_arr(i,j,k) = SumOfIntegratedPotential2D(x_hi-x, y,      dx, dy); }
+                } else {
+                    int const k0 = k - lo[2];
+                    amrex::Real const z = k0*dz;
```

Comment:
Could we introduce `z_hi` at this point?

Reply:
Yes, but that would be inside a loop over `i,j,k`. 
I thought that it would be better to define it beforehand, just once, since it does not depend on either `i`, `j` or `k`.